include/ccf/node_context.h

```diff
@@ -3,19 +3,21 @@
 #pragma once
 
 #include "ccf/historical_queries_interface.h"
+#include "ccf/host_processes_interface.h"
 #include "ccf/indexing/indexer_interface.h"
 #include "ccf/indexing/lfs_interface.h"
-#include "node/rpc/node_interface.h"
 
 namespace ccfapp
 {
   struct AbstractNodeContext
   {
     virtual ~AbstractNodeContext() = default;
 
+    virtual ccf::NodeId get_node_id() const = 0;
     virtual ccf::historical::AbstractStateCache& get_historical_state() = 0;
-    virtual ccf::AbstractNodeState& get_node_state() = 0;
     virtual ccf::indexing::IndexingStrategies& get_indexing_strategies() = 0;
     virtual ccf::indexing::AbstractLFSAccess& get_lfs_access() = 0;
+
+    virtual ccf::AbstractHostProcesses* get_host_processes() = 0;
```

Comment:
Since we want to be able to launch host processes from JS generic, we currently need to make this a public thing that _any_ app could do. My generic-container solution would avoid this - JS generic is privately implemented, so may make use of framework-private subsystems that aren't exposed to any other app. I think that's an interesting win, though not something we should do _too_ often.